welly/las.py

```diff
@@ -0,0 +1,366 @@
+    path = utils.to_filename(path)
+
+    # instantiate new LASFile to parse data and header to
+    las = lasio.LASFile()
+
+    # unpack datasets
+    for dataset, (data, header) in datasets.items():
+
+        if dataset == 'Curves':
+            # parse header pd.DataFrame to LASFile
+            for section_name in set(header.section.values):
+
+                # get header section
+                df_section = header[header.section == section_name]
+
+                if section_name == 'Version':
+                    if len(df_section[
+                               df_section.original_mnemonic == 'VERS']) > 0:
+                        las.version.VERS = \
+                            df_section[df_section.original_mnemonic == 'VERS'][
+                                'value'].values[0]
+                    if len(df_section[
+                               df_section.original_mnemonic == 'WRAP']) > 0:
+                        las.version.WRAP = \
+                            df_section[df_section.original_mnemonic == 'WRAP'][
+                                'value'].values[0]
+                    if len(df_section[
+                               df_section.original_mnemonic == 'DLM']) > 0:
+                        las.version.DLM = \
+                            df_section[df_section.original_mnemonic == 'DLM'][
+                                'value'].values[0]
+
+                elif section_name == 'Well':
+                    las.sections["Well"] = SectionItems(
+                        [HeaderItem(r.original_mnemonic, r.unit, r.value,
+                                    r.descr) for i, r in
+                         df_section.iterrows()])
+
+                elif section_name == 'Parameter':
+                    las.sections["Parameter"] = SectionItems(
+                        [HeaderItem(r.original_mnemonic, r.unit, r.value,
+                                    r.descr) for i, r in
+                         df_section.iterrows()])
+
+                elif section_name == 'Curves':
+                    for i, header_row in df_section.iterrows():
+                        if header_row.mnemonic in data.columns:
+                            curve_data = data.loc[:, header_row.mnemonic]
+                            las.append_curve(mnemonic=header_row.mnemonic,
+                                             data=curve_data,
+                                             unit=header_row.unit,
+                                             descr=header_row.descr,
+                                             value=header_row.value)
+
+                elif section_name == 'Other':
+                    las.sections["Other"] = df_section['descr'][0]
+
+                else:
+                    warnings.warn(
+                        'Section was not recognized and not parsed: "{}"'.format(
+                            section_name))
+        else:
+            raise NotImplementedError('LAS 3.0 is not yet supported.')
```

Comment:
Not sure how this comment come here. You don't specify .las version to write to in to_las function. Is that something recognized by presence or absence of "Curves".  and how do we know it is las 3.0 and not some other problem?

Reply:
LAS 1.2 and 2.0 only have '~Curve' data section. LAS 3 can have:

> ~Ascii or ~Log
> ~Core
> ~Inclinometry
> ~Drilling
> ~Tops
> ~Test

Still have to investigate if we can read them in the same way as ~Curve. So for now we only parse ~Curve datasets. Added a TODO for this.